### ISSUE TITLE
FP and FNs for documentbuilderfactory-disallow-doctype-decl-missing:

### ISSUE THREAD

**Issue by sarr424** [returntocorp/sgrep-rules#3831]
Hi, I found some pattern-coverage gaps in the Java XXE rule documentbuilderfactory-disallow-doctype-decl-missing.

Case 1 (FP): pattern-sanitizers only recognize setFeature(...). Needs setAttribute support:
- patterns:
    - pattern: $FACTORY.setAttribute($ATTR, "")
    - metavariable-regex:
        metavariable: $ATTR
        regex: ACCESS_EXTERNAL_DTD|ACCESS_EXTERNAL_SCHEMA
Cases 2 & 3 (FN): pattern-sources requires $FACTORY = DocumentBuilderFactory.newInstance(). 
Needs a supplementary source for chained calls:
- pattern: DocumentBuilderFactory.newInstance().newDocumentBuilder()
- pattern: $X = DocumentBuilderFactory.newInstance().newDocumentBuilder()

```java
package example;
import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.ParserConfigurationException;
import javax.xml.XMLConstants;
import org.xml.sax.SAXException;
import java.io.IOException;
class AllCases {
    // ── Case 1: FP · setAttribute-based defense ──
    // Should NOT be flagged (ACCESS_EXTERNAL_DTD="" blocks external entities).
    // Semgrep flags it anyway → FALSE POSITIVE
    public void case1_setAttribute() throws ParserConfigurationException {
        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
        dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
        dbf.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
        dbf.setAttribute(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
        dbf.newDocumentBuilder();
    }
    // ── Case 2: FN · full chained call ──
    // Should be flagged (no XXE protection at all).
    // Semgrep misses it → FALSE NEGATIVE
    public void case2_fullChained() throws ParserConfigurationException, SAXException, IOException {
        DocumentBuilderFactory.newInstance().newDocumentBuilder().parse("system.xml");
    }
    // ── Case 3: FN · inline chained assignment ──
    // Should be flagged (no XXE protection at all).
    // Semgrep misses it → FALSE NEGATIVE
    public void case3_inlineChained() throws ParserConfigurationException {
        DocumentBuilder db = DocumentBuilderFactory.newInstance().newDocumentBuilder();
    }
}
```

**Expected behavior**

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Additional Context**
